src/generate.js

```diff
@@ -186,31 +190,25 @@ export const generateCSS = (
  * See generateCSSRuleset for usage and documentation of paramater types.
  */
 const runStringHandlers = (
-    declarations /* : SheetDefinition */,
+    declarations /* : OrderedElements */,
     stringHandlers /* : StringHandlers */,
     selectorHandlers /* : SelectorHandler[] */
 ) /* */ => {
-    const result = {};
```

Comment:
:( I undid some of your for-loop optimization here, but putting it back is lot uglier now with the `OrderedElements` thing. I re-did some of the optimizations inside of the `OrderedElements` implementation, though.